graspy/align/base.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from abc import abstractmethod
+from sklearn.utils import check_array
+from sklearn.base import BaseEstimator
+
+
+class BaseAlign(BaseEstimator):
+    """
+    Base class for align tasks such as sign flipping, procrustes and seedless
+    procrustes.
+
+    Attributes
+    ----------
+        Q_ : array, size (d, d)
+              final orthogonal matrix, used to modify X.
+
+    """
+
+    def __init__(self):
+        pass
+
+    def _check_datasets(self, X, Y):
+        """
+        Ensures that the datasets are numpy, 2d, finite, and have the same
+        number of components. Does not check for same number of vertices.
+        Returns copies of these datasets.
+        """
+        # check for numpy-ness
+        if not isinstance(X, np.ndarray):
+            msg = f"First dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        if not isinstance(Y, np.ndarray):
+            msg = f"Second dataset is a {type(Y)}, not an np.ndarray! "
+            raise TypeError(msg)
+        # check for 2-dness and finiteness
+        X = check_array(X, copy=True)
+        Y = check_array(Y, copy=True)
+        # check for equal components
+        if X.shape[1] != Y.shape[1]:
+            msg = "Two datasets have different number of components!"
+            raise ValueError(msg)
+        return X, Y
+
+    @abstractmethod
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn the matrix Q_ that aligns the first
+        dataset with the second.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
```

Comment:
I know you also have in the first section of docstring, but i think worth specifying that X is the one being mapped to Y somehow in this part. comment applies throughout. thoughts?

Reply:
i understand the comment, but so far hasn't been able to come up with a clear and concise way of specifying that an additional time.

Reply:
If it makes it easier, in my priority list I have:
1. Clear and concise
2. Clear and long
3. Long
4. Concise

3 and 4 are significantly less prized than 3 or 4 though

Reply:
(this one im still thinking about)

Reply:
X : Dataset to be mapped to Y, must have same number of dimensions (axis 1) as Y.
Y : Target dataset, must have same number of dimensions (axis 1) as X.

Reply:
From scipy 

`A(M, N) array_like
Matrix to be mapped.

B(M, N) array_like
Target matrix.`

also seems fine?

Reply:
i like your first option, second seems almost 'too concise'. my only real issue with it is: in fit_predict this makes total sense, but in fit, we are only learning the transformation, not actually mapping Y. am i overthinking?

Reply:
i think it makes sense even for fit personally

Reply:
in that case, i'll just do that.